src/screens/InitWallet.js

```diff
@@ -235,9 +235,53 @@ class NewWordsScreen extends React.Component {
   }
 }
 
+/**
+ * Handles the validation of the raw seed words input
+ * @param {string} rawWords String containing all the words written by the user
+ * @returns {{words: string[], errorMessage: string, isValid: boolean}}
+ */
+function handleValidWords(rawWords) {
+  let trimmedWordsStr = rawWords.trim(/\s+/);
+  let errorMessage = '';
+  let isValid = false;
+
+  try {
+    const { valid, words } = walletUtils.wordsValid(trimmedWordsStr);
+    trimmedWordsStr = words;
+    isValid = valid;
+  } catch (e) {
+    // Handling unknown errors
+    if (!(e instanceof hathorErrors.InvalidWords)) {
+      throw e;
+    }
+
+    // Handling Invalid Words errors
+    isValid = false;
+    errorMessage = e.message;
+    const listOfInvalidWords = e.invalidWords.filter((word) => word.length !== 0);
+    if (listOfInvalidWords.length > 0) {
+      errorMessage = `${errorMessage} List of invalid words: ${e.invalidWords.join(', ')}.`;
+    }
+  }
+  const wordsArr = trimmedWordsStr.split(/\s+/);
+  const nonEmptyWords = wordsArr.filter((word) => word.length !== 0);
+  const lowercaseWords = nonEmptyWords.map((word) => word.toLowerCase());
+
+  return {
+    words: lowercaseWords,
+    errorMessage,
+    isValid,
+  }
+}
+
 class LoadWordsScreen extends React.Component {
+  /**
+   * State for the import wallet words screen
+   * @type {{rawWords: '', words: string[], errorMessage: string, isValid: boolean}}
+   */
   state = {
-    words: '',
+    rawWords: '',
+    words: [],
```

Comment:
Are you using this state?

Reply:
Yes, the `handleValidWords()` method requires the raw text from the TextArea. For the `onChangeText()` event handler that's trivial, because it receives it at each change. For the `loadClicked()`, however, I would need to somehow send it.

The cleanest solution I found was to always update the `rawWords` at each change. The alternative would be to send an event on the `loadClicked()` events that were raised by the submit button only, and that felt cumbersome.